clang/lib/AST/ASTContext.cpp

```diff
@@ -14526,6 +14564,9 @@ static QualType getCommonSugarTypeNode(ASTContext &Ctx, const Type *X,
                                       DX->isCountInBytes(), DX->isOrNull(),
                                       CDX);
   }
+  case Type::PredefinedSugar:
+    // FIXME: Should this be reachable here?
+    return QualType();
```

Comment:
```suggestion
  case Type::PredefinedSugar:
    assert(cast<PredefinedSugarType>(X)->getKind() != cast<PredefinedSugarType>(Y)->getKind());
    return QualType();
```

Yeah, this can be reached if you have two PredefinedTypes which have the same canonical type.

The assert is helpful because if you reach here and they have the same kind, that means we screwed up the uniquing and created two different PredefinedTypes with the same kind.